pkg/git/repository.go

```diff
@@ -283,16 +294,12 @@ func (r *repo) CurrentBranch() (string, error) {
 	return ref.Name().Short(), nil
 }
 
-func (r *repo) commit(opts *PushOptions) (*plumbing.Hash, error) {
+func (r *repo) commit(ctx context.Context, opts *PushOptions) (*plumbing.Hash, error) {
 	var h plumbing.Hash
 
-	cfg, err := r.ConfigScoped(config.SystemScope)
+	author, err := r.getAuthor(ctx, opts.Provider)
```

Comment:
passing `opts.Provider` here does nothing since you already have your provider instance on `opts.provider`